Assets/Mirage/Runtime/MessageHandler.cs

```diff
@@ -21,7 +21,7 @@ public class MessageHandler : IMessageReceiver
         /// <param name="reader"></param>
         internal delegate void NetworkMessageDelegate(INetworkPlayer player, NetworkReader reader);
 
-        internal readonly Dictionary<int, NetworkMessageDelegate> messageHandlers = new Dictionary<int, NetworkMessageDelegate>();
+        internal readonly Dictionary<int, HashSet<NetworkMessageDelegate>> messageHandlers = new Dictionary<int, HashSet<NetworkMessageDelegate>>();
```

Comment:
this can be a normal list instead of HashSet because each added handler is wrapped anyway so will be unique (even if added same method twice)